### ISSUE TITLE
Routing is not working in this example

### ISSUE THREAD

**Issue by shamraj90** [ExampleDriven/spring-boot-docker-example#1]
I have imported this app run it in docker but the api gateway (routing) is not working. Below is the exception-
2016-09-21 07:22:21.087  WARN 1 --- [nio-9090-exec-1] o.s.c.n.zuul.web.ZuulHandlerMapping      : No routes found from RouteLocator

**Comment by szantopeter** [comment]
Have you started up all three components using the 

``` sh
docker-compose up
```

command? If so then what services do you see when you look at the eureka console

```
http://<<docker-host>>:8761
```

**Comment by shamraj90** [comment]
I have started individually by linking the containers using the command as mentioned in the blog

**Comment by shamraj90** [comment]
only the echo service is available on eureka console

**Comment by szantopeter** [comment]
in this case most likely something is wrong with the way you linked the docker images. Echo service seems to be linked with eureka, but zuul don't. Either try restarting the zuul image and see if it helps or use docker-compose it is better anyway...

**Comment by shamraj90** [comment]
Ok thanks for the info.. Will try and let you know

**Comment by shamraj90** [comment]
When i use docker compose command its working. Thanks

**Comment by shamraj90** [comment]
And can we run this example without using eureka?.. If so can you help in running this example?

**Comment by szantopeter** [comment]
This example is built with eureka, but you can start up echo service without eureka, just remove the eureka dependency from it.

**Comment by shamraj90** [comment]
Does it work if i remove only eureka in docker-compose file?

**Comment by szantopeter** [comment]
if you remove eureka from docker compose then it won't start up, but the other two services are still looking for it. You need to modify them to not to look for eureka

**Comment by shamraj90** [comment]
Hello i have an another query.. Where it is configured in zuul ,so that it finds echo service? Its not configured in application.yml. It has defined only prefix:api. And where i define to route to another service like echo service 2

**Comment by shamraj90** [comment]
For example: if requests is like api/echoservice it should route to echoservice container and if request is like api/echoservice2 , it should route to echoservice2 container

**Comment by szantopeter** [comment]
See here https://exampledriven.wordpress.com/2016/07/06/spring-cloud-zuul-example/
and the official spring cloud zuul documentation

**Comment by shamraj90** [comment]
Thank you so much :)

**Comment by shamraj90** [comment]
The example which u had sent involves eureka server, that's fine.. I have tried earlier example removing eureka server but am not able to access the app. In the console it says-No routes found from RouteLocator. And in the browser, following is displayed-

Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this as a fallback.

Thu Sep 22 06:43:06 UTC 2016
There was an unexpected error (type=Not Found, status=404).
No message available

Where i have gone wrong? If i include eureka server it works fine.

**Comment by shamraj90** [comment]
Hello szantopeter https://github.com/szantopeter,

since the issue was closed am mailing you because am facing this issue
without eureka so kindly help me in this.

I have tried the example removing eureka server but am not able to access
the app. In the console it says-No routes found from RouteLocator. And in
the browser, following is displayed -

Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this
as a fallback.

Thu Sep 22 06:43:06 UTC 2016
There was an unexpected error (type=Not Found, status=404).
No message available

Where i have gone wrong? If i include eureka server it works fine.
